src/app/containers/StakePage/components/FeeBlock.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useCallback, useMemo, useEffect } from 'react';
+import { useCachedAssetPrice } from 'app/hooks/trading/useCachedAssetPrice';
+import { useAccount } from 'app/hooks/useAccount';
+import { useTranslation } from 'react-i18next';
+import { AssetDetails } from 'utils/models/asset-details';
+import { getContract } from 'utils/blockchain/contract-helpers';
+import { Tooltip } from '@blueprintjs/core';
+import { numberFromWei, weiTo4 } from 'utils/blockchain/math-helpers';
+import {
+  staking_withdrawFee,
+  staking_numTokenCheckpoints,
+} from 'utils/blockchain/requests/staking';
+import { useStaking_getAccumulatedFees } from 'app/hooks/staking/useStaking_getAccumulatedFees';
+import { translations } from 'locales/i18n';
+import { LoadableValue } from 'app/components/LoadableValue';
+import { ContractName } from 'utils/types/contracts';
+import { bignumber } from 'mathjs';
+import { Asset } from 'types';
+import { weiToUSD } from 'utils/display-text/format';
+
+interface IFeeBlockProps {
+  contractToken: AssetDetails;
+  usdTotal: (asset: AssetDetails, value: number) => void;
```

Comment:
This is bad naming, when I looked at the `useEffect` where it's used, I thought you are updating some variable but this is a callback. Name it more appropriate (e.g. `updateUsdTotal`).